app/extensions/edm/__init__.py

```diff
@@ -200,6 +200,28 @@ def request_passthrough_result(
         )
         return result
 
+    def transfer_data_setting(self):
```

Comment:
This shoudl be in the invoke task not here. Then as much of the transfer logic as possible is all in one place